yt_dlp/downloader/hls.py

```diff
@@ -307,13 +313,82 @@ def download_fragment(fragment):
 
                 return frag_content, frag_index
 
+            pack_fragment = lambda frag_content, _: frag_content
+
+            if is_webvtt:
+                def pack_fragment(frag_content, frag_index):
+                    output = io.StringIO()
+                    adjust = 0
+                    for block in webvtt.parse_fragment(frag_content):
+                        if isinstance(block, webvtt.CueBlock):
+                            block.start += adjust
+                            block.end += adjust
+
+                            dedup_window = extra_state.setdefault('webvtt_dedup_window', [])
+                            cue = block.as_json
+
+                            # skip the cue if an identical one appears
+                            # in the window of potential duplicates
+                            # and prune the window of unviable candidates
+                            i = 0
+                            skip = True
+                            while i < len(dedup_window):
+                                window_cue = dedup_window[i]
+                                if window_cue == cue:
+                                    break
+                                if window_cue['end'] >= cue['start']:
+                                    i += 1
+                                    continue
+                                del dedup_window[i]
+                            else:
+                                skip = False
+
+                            if skip:
+                                continue
+
+                            # add the cue to the window
+                            dedup_window.append(cue)
+                        elif isinstance(block, webvtt.Magic):
+                            # take care of MPEG PES timestamp overflow
+                            if block.mpegts is None:
+                                block.mpegts = 0
+                            extra_state.setdefault('webvtt_mpegts_adjust', 0)
+                            block.mpegts += extra_state['webvtt_mpegts_adjust'] << 33
+                            if block.mpegts < extra_state['webvtt_mpegts_last']:
+                                extra_state['webvtt_mpegts_adjust'] += 1
+                                block.mpegts += 1 << 33
+                            extra_state['webvtt_mpegts_last'] = block.mpegts
+
+                            if frag_index == 1:
+                                extra_state['webvtt_mpegts'] = block.mpegts or 0
+                                extra_state['webvtt_local'] = block.local or 0
+                                # XXX: block.local = block.mpegts = None ?
+                            else:
+                                if block.mpegts is not None and block.local is not None:
+                                    adjust = (
+                                        (block.mpegts - extra_state.get('webvtt_mpegts', 0)) -  # noqa: W504
```

Comment:
I almost formatted this expression as
```python
(block.mpegts - extra_state.get('webvtt_mpegts', 0)) -
(block.local  - extra_state.get('webvtt_local',  0))
```
to emphasise the symmetry between the two subtractions. But I figured I can live with just a couple spaces of misalignment. Moving the operator destroys it completely, however.

The reason is that this is a stupid warning that should not exist. Telling me how many and what kind of spaces I am allowed to put between operators is useless pedantry that actively makes the code worse.

Also, the recommendation [used to be just the opposite](https://www.flake8rules.com/rules/W503.html).

> Anti-pattern
>
> Note: Despite being in the anti-pattern section, this will soon be considered the best practice.

If they can’t make their minds up about what the best practice is, how am I supposed to treat them seriously?

Reply:
For one, I don't think this PEP makes the code any less readable. For me atleast,
```py
(block.mpegts - extra_state.get('webvtt_mpegts', 0))
- (block.local - extra_state.get('webvtt_local', 0))
```
is just as readable as
```py
(block.mpegts - extra_state.get('webvtt_mpegts', 0)) -
(block.local  - extra_state.get('webvtt_local',  0))
```

I personally prefer this instead.
```py
  (block.mpegts - extra_state.get('webvtt_mpegts', 0))
- (block.local  - extra_state.get('webvtt_local',  0))
```

But if every developer working on this code decide to style the code how they think is best, there will soon be no consistent styling anywhere. So unless there is a significant benefit in changing it, the flake8 rules must be obeyed